source/common/network/connection_impl.cc

```diff
@@ -539,6 +540,12 @@ void ConnectionImpl::doConnect() {
       ENVOY_CONN_LOG(debug, "immediate connection error: {}", *this, errno);
     }
   }
+
+  try {
+    local_address_ = Address::addressFromFd(fd_);
+  } catch (const EnvoyException& e) {
+    // Eat the exception and leave local_address unchanged
```

Comment:
nit: period end of sentence. Also, can you add more comments on when this would actually happen? AFAICT this is for some cases of UDS?